### PR TITLE
DB-8036 allow more predicates to be pushed down into unions

### PR DESCRIPTION
**Pull request by yxia92**
Please refer to [DB-8036](https://splicemachine.atlassian.net/browse/DB-8036) for details.

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java [hunk 1]
```diff
@@ -320,40 +320,22 @@ public void printSubNodes(int depth){
...
     /**
-     * Find colName in the result columns and return underlying columnReference.
-     * Note that this function returns null if there are more than one FromTable
-     * for this SelectNode and the columnReference needs to be directly under
-     * the resultColumn. So having an expression under the resultSet would cause
-     * returning null.
-     *
-     * @param    colName        Name of the column
-     * @return ColumnReference    ColumnReference to the column, if found
+     * Remap the column reference node to underlying column reference in the select, if possible.
```
Comment (jaceklaskowski) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java]: Does the method really "remap" or simply gives a source column reference? The name suggests the latter.

Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java]: Thanks @jaceklaskowski for the review. "Remap" here I meant to "find the corresponding source column reference". I updated the comment to make the comment and function name consistent.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java [hunk 2]
```diff
@@ -320,8 +320,8 @@ public void printSubNodes(int depth){
     public FromList getFromList(){ return fromList; }
 
     /**
-     * Remap the column reference node to underlying column reference in the select, if possible.
-     * The select node here corresponds to a union branch
+     * Given a column reference, find the corresponding column reference in the select, if possible.
```
Comment (jaceklaskowski) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java]: I'd word it as follows:

```
Finds the column reference (as used in select) for a given column reference
```

The rest is in `@param` and `@return`

Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java]: Thanks. I've updated the comment accordingly.

**Pull request merged**